clang/include/clang/Basic/TargetInfo.h

```diff
@@ -768,10 +768,10 @@ class TargetInfo : public virtual TransferrableTargetInfo,
 
   /// Return the "preferred" register width on this target.
   virtual unsigned getRegisterWidth() const {
-    // Currently we assume the register width on the target matches the pointer
+    // Currently we assume the register width on the target matches the size
     // width, we can introduce a new variable for this if/when some target wants
     // it.
-    return PointerWidth;
+    return getTypeWidth(SizeType);
```

Comment:
Possibly worth upstreaming?

Reply:
Not sure, depends if we end up changing our minds on what this should do, and it's not like we're going to get conflicts for this very often...

Reply:
Yeah probably not worth the effort right now.